src/pages/Invoice/CreateInvoice.js

```diff
@@ -1,7 +1,376 @@
-import React from "react";
+/* eslint-disable no-unused-vars */
+import React, { Fragment, useState, useEffect } from "react";
+import withWrapper from "common/withWrapper";
+import Navbar from "common/Navbar";
+import "styles/add-invoice.css";
+import { Link, useHistory } from "react-router-dom";
+import ChangeCustomerModal from "./ChangeCustomerModal";
+import AddItemModal from "./AddItemsModal";
+import { FormattedMessage } from "react-intl";
+import { useNotification } from "notification";
 
 const CreateInvoice = () => {
-    return <div>I am add customers page</div>;
+    const [itemModal, setItemModal] = useState(false);
+    const [customerDetailsModal, setCustomersDetailsModal] = useState(false);
+    const [customersInfo, setCustomersInfo] = useState([]);
+    const [itemInfo, setItemInfo] = useState([]);
+    const [invoiceRecipentDetails, setInvoiceRecipentDetails] = useState({
+        name: "",
+        phone: "",
+        email: "",
+        issueDate: "",
+        dueDate: "",
+        invoiceNumber: "",
+        referenceNumber: "",
+        items: [],
+        notes: "",
+    });
+    const history = useHistory();
+    const { triggerNotification } = useNotification();
+
+    useEffect(() => {
+        fetchData();
+    }, []);
+    // Function to fetch data from local storage
+    const fetchData = () => {
+        if (localStorage.getItem("customer_data"))
+            setCustomersInfo(JSON.parse(localStorage.getItem("customer_data")));
+        if (localStorage.getItem("inventory_data"))
+            setItemInfo(JSON.parse(localStorage.getItem("inventory_data")));
+    };
+    // Function to delete items
+    const removeElement = (id) => {
+        if (window.confirm("Are you sure, you eant to delete this item?")) {
+            const items = invoiceRecipentDetails.items.filter(
+                (item) => item.id !== id
+            );
+            setInvoiceRecipentDetails({ ...invoiceRecipentDetails, items });
```

Comment:
```suggestion
            setInvoiceRecipentDetails((invoiceDetail)=>({ ...invoiceRecipentDetails, items: invoiceDetail.items.filter(item=>item.id !== id) }));
```
use callback function of state